llama_index/program/openai_program.py

```diff
@@ -33,6 +40,39 @@ def _get_json_str(raw_str: str, start_idx: int) -> Tuple[Optional[str], int]:
     return None, start_idx
 
 
+def _parse_tool_calls(
+    tool_calls: List[dict],
+    output_cls: Type[BaseModel],
+    allow_multiple: bool = False,
+    verbose: bool = False,
+) -> Union[BaseModel, List[BaseModel]]:
+    outputs = []
+    for tool_call in tool_calls:
+        function_call = tool_call["function"]
+        if verbose:
+            name = function_call["name"]
+            arguments_str = function_call["arguments"]
+            print(f"Function call: {name} with args: {arguments_str}")
+
+        if isinstance(function_call["arguments"], dict):
```

Comment:
ooc when would it be one or the other

Reply:
this is existing code we have. I think it was due to difference in OpenAI and AzureOpenAI behavior in the past.